Revert "dbus_config: Remove dbus configuration file"

martin-ejdestig opened the pull request:
This reverts commit 103be9f0bb77df0b139541ab366ff98a4fe4e408.
    
No D-Bus policy file is installed which causes persistence-administrator to fail during startup since it ends up calling a persitence-client-object function that wants to own the name `org.genivi.persistence.admin`. Owning names is denied by default on the system bus.
    
Since it is persistence-common-object that takes the name org.genivi.persistence.admin in `src/pers_ipc_dbus.c:persIpcPASLoopThread()`, the configuration probably belongs in this repository.
    
It looks like persistence-client-library has never installed `org.genivi.persistence.admin.conf` itself in its build system. It was done for a while in meta-ivi which must have been the source of the conflict
mentioned in the reverted commit. Installed in GENIVI/meta-ivi@7318681d851e3e3ee22a but later removed in GENIVI/meta-ivi@9d5a45e473fbb3a29f5e with the comment "it should be done
by PCO".
    
The reverted commit was sent in pull request #3 with no reasoning as to why it should be added by persistent-client-library instead of persistence-common-object. Later pull request #5 first adds it but then removes it again in a later commit. Unsure why. Reasoning in comment about the missing D-Bus config file seemed correct.

martin-ejdestig committed:
Revert "dbus_config: Remove dbus configuration file"

This reverts commit 103be9f0bb77df0b139541ab366ff98a4fe4e408.

No D-Bus policy file is installed which causes persistence-administrator
to fail during startup since it ends up calling a persitence-client-object
function that wants to own the name org.genivi.persistence.admin. Owning
names is denied by default on the system bus.

Since it is persistence-common-object that takes the name
org.genivi.persistence.admin in src/pers_ipc_dbus.c:persIpcPASLoopThread(),
the configuration probably belongs in this repository.

It looks like persistence-client-library has never installed
org.genivi.persistence.admin.conf itself in its build system. It was done
for a while in meta-ivi which must have been the source of the conflict
mentioned in the reverted commit. Installed in
GENIVI/meta-ivi@7318681d851e3e3ee22a but later removed in
GENIVI/meta-ivi@9d5a45e473fbb3a29f5e with the comment "it should be done
by PCO".

The reverted commit was sent in pull request #3 with no reasoning as to why
it should be added by persistent-client-library instead of
persistence-common-object. Later pull request #5 first adds it but then
removes it again in a later commit. Unsure why. Reasoning in comment about
the missing D-Bus config file seemed correct.

Signed-off-by: Martin Ejdestig <mejdestig@luxoft.com>

martin-ejdestig commented:
Recipes in GENIVI/meta-ivi should of course be updated to install the file again when this has been merged.

martin-ejdestig commented:
Please also note GENIVI/persistence-client-library#16 . Should not be merged before a decision has been taken here that the policy file really belongs in this repository. Want to solve this once and for all... not go through another round (ot two) of "add and revert". ;)

gunnarx commented:
I understand your description.  This repository now lacks official maintainer (someone should volunteer to do it if they are using the software...).   But I can help you merge this here, and in meta-ivi also.

> Recipes in GENIVI/meta-ivi should of course be updated to install the file again when this has been merged.

That would be the recipe for persistence-common-object that needs updating then?

martin-ejdestig commented:
> That would be the recipe for persistence-common-object that needs updating then?

Yes, it currently only puts files in ${prefix} in the package and the D-Bus policy file is installed in ${sysconfdir}. Recipe actually does "rm -rf ${sysconfdir}" to work around previous conflict that the reverted commit in this PR also tried to solve. See:

https://github.com/GENIVI/meta-ivi/blob/0a0aeecdfeb61f983c8c02c264162527ee8fb51d/meta-ivi/recipes-extended/persistence-common-object/persistence-common-object_1.1.0.bb#L23

martin-ejdestig committed:
Install D-Bus policy file in /usr/share/dbus-1/system.d

Instead of /etc/dbus-1/system.d. Package supplied policy configuration
in /etc/dbus-1/system.d is officially deprecated in D-Bus since 1.13.10 and
is reserved for system administrators wanting to override default behavior.
/usr/share/dbus-1/system.d has been preferred since D-Bus 1.10.0 (released
~4 years ago). See the following commit in D-Bus for deprecation notice:

https://gitlab.freedesktop.org/dbus/dbus/commit/dee0f551112349da4aa3f913f92f2c7c8f48f1f0

Signed-off-by: Martin Ejdestig <mejdestig@luxoft.com>

martin-ejdestig commented:
Put another commit on top that installs the policy file in `/usr/share/dbus-1/system.d` instead of `/etc/dbus-1/system.d` instead of temporarily working around it in a bbappend. See commit message for details as to why it makes sense to change directory (regardless of meta-ivi recipe rm:ifery).

gunnarx commented:
Seems legit. :)  What would be the final change to meta-ivi?  Just remove the "rm:ifery" and no special install steps.   The standard autotools yocto class should then **make install** and this will now include the d-bus file in the right place?  (That's a quick fix I could make but if something more advanced is needed, could you send a meta-ivi patch to help?)

martin-ejdestig commented:
Jupp, only need to bump git hash and cleanup the rm:ifery in meta-ivi recipe.

gunnarx commented:
Actually, this can be merged now without issue since meta-ivi components are all on fixed commit hashes of course.  I'm ready to pull the trigger - or any more commits coming?

martin-ejdestig commented:
No more commits for this pr. The last one was what I needed to have the pas daemon at least start on target.
